crates/accelerate/src/gate_direction.rs

```diff
@@ -142,8 +157,505 @@ where
     Ok(true)
 }
 
+//#########################################################################
+//              GateDirection transformation pass functions
+//#########################################################################
+
+/// Try to swap two-qubit gate directions using pre-defined mapping to follow the right direction with respect to the coupling map.
+///
+/// Args:
+///     dag: the DAGCircuit to analyze
+///
+///     coupling_edges: set of edge pairs representing a directed coupling map, against which gate directionality is checked
+///
+/// Returns:
+///     the transformed DAGCircuit
+#[pyfunction]
+#[pyo3(name = "fix_gate_direction_coupling")]
+fn py_fix_with_coupling_map(
+    py: Python,
+    dag: &mut DAGCircuit,
+    coupling_edges: HashSet<[Qubit; 2]>,
+) -> PyResult<DAGCircuit> {
+    if coupling_edges.is_empty() {
+        return Ok(dag.clone());
+    }
+
+    let coupling_map_check =
+        |_: &PackedInstruction, op_args: &[Qubit]| -> bool { coupling_edges.contains(op_args) };
+
+    fix_gate_direction(py, dag, &coupling_map_check, None).cloned()
+}
+
+/// Try to swap two-qubit gate directions using pre-defined mapping to follow the right direction with respect to the given target.
+///
+/// Args:
+///     dag: the DAGCircuit to analyze
+///
+///     coupling_edges: set of edge pairs representing a directed coupling map, against which gate directionality is checked
+///
+/// Returns:
+///     the transformed DAGCircuit
+#[pyfunction]
+#[pyo3(name = "fix_gate_direction_target")]
+fn py_fix_with_target(py: Python, dag: &mut DAGCircuit, target: &Target) -> PyResult<DAGCircuit> {
+    let target_check = |inst: &PackedInstruction, op_args: &[Qubit]| -> bool {
+        let qargs = smallvec![
+            PhysicalQubit::new(op_args[0].0),
+            PhysicalQubit::new(op_args[1].0)
+        ];
+
+        // Take this path so Target can check for exact match of the parameterized gate's angle
+        if let OperationRef::Standard(std_gate) = inst.op.view() {
+            match std_gate {
+                StandardGate::RXXGate
+                | StandardGate::RYYGate
+                | StandardGate::RZZGate
+                | StandardGate::RZXGate => {
+                    return target
+                        .py_instruction_supported(
+                            py,
+                            None,
+                            Some(qargs),
+                            Some(
+                                get_std_gate_class(py, std_gate)
+                                    .expect("These gates should have Python classes")
+                                    .bind(py),
+                            ),
+                            Some(inst.params_view().to_vec()),
+                        )
+                        .unwrap_or(false)
+                }
```

Comment:
For a follow up, or maybe even for #13308, we should be able to test instruction compatibility with a `Target` from a `StandardGate`.

Reply:
Should be left as is for now. We can address it in a follow up.